In [1]:
import numpy as np
import pandas as pd

In [2]:
# The statistical module used to run chi square test
import scipy.stats as stats

In [4]:
# Observed data in a (hypothetical) survey of 300 Dinosaurs 
observed = pd.Series([220,55,25], index=["omnivores", "carnivores", "herbivores"])

In [14]:
pd.DataFrame({'Type':['omnivores','carnivores','herbivores']})

Type
0   omnivores
1  carnivores
2  herbivores

In [15]:
# Create a data frame
df = pd.DataFrame([observed])

In [16]:
d

0
omnivores   220
carnivores   55
herbivores   25

In [6]:
# Add a column whose default values are the expected values
df[1] = 100

In [7]:
# Rename columns
df.columns = ["observed", "expected"]

In [8]:
# View the data frame
df

observed  expected
omnivores        220       100
carnivores        55       100
herbivores        25       100

In [9]:
# The degree of freedom is 3-1 = 2
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 2)

In [10]:
# The critical value
critical_value

5.991464547107979

In [10]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected'])

Power_divergenceResult(statistic=220.5, pvalue=1.3153258948574585e-48)

### Conclusion

* Since the chi square value of 220.5 exceeds the critical value of 5.99, we conclude that the results are statistically significant.